pysatMissions/instruments/missions_sgp4.py

```diff
@@ -118,93 +169,161 @@ def load(fnames, tag=None, inst_id=None, obs_long=0., obs_lat=0., obs_alt=0.,
+
+    # If provided, use user-specified TLEs.  Otherwise use ISS defaults above.
     if TLE1 is not None:
         line1 = TLE1
     if TLE2 is not None:
         line2 = TLE2
 
-    if num_samples is None:
-        num_samples = 100
-
-    # Create satellite from TLEs and assuming a gravity model;
-    # according to module webpage, wgs72 is common
-    satellite = twoline2rv(line1, line2, wgs72)
+    if (num_samples is None) or one_orbit:
+        num_samples = 86400
 
     # Extract list of times from filenames and inst_id
     times, index, dates = ps_meth.generate_times(fnames, num_samples,
                                                  freq=cadence)
-
-    # Create list to hold satellite position, velocity
-    position = []
-    velocity = []
-    for timestep in index:
-        # orbit propagator - computes x,y,z position and velocity
-        pos, vel = satellite.propagate(timestep.year, timestep.month,
-                                       timestep.day, timestep.hour,
-                                       timestep.minute, timestep.second)
-        position.extend(pos)
-        velocity.extend(vel)
+    # Calculate epoch for orbital propagator
+    epoch_days = (epoch - dt.datetime(1949, 12, 31)).days
+    jd, _ = sapi.jday(dates[0].year, dates[0].month, dates[0].day, 0, 0, 0)
+
+    if inclination is not None:
+        # If an inclination is provided, specify by Keplerian elements
+        eccentricity, mean_motion = orbits.convert_to_keplerian(alt_periapsis,
+                                                                alt_apoapsis)
+        satellite = sapi.Satrec()
+        # according to module webpage, wgs72 is common
+        satellite.sgp4init(sapi.WGS72, 'i', 0, epoch_days, bstar, 0, 0,
+                           eccentricity, np.radians(arg_periapsis),
+                           np.radians(inclination), np.radians(mean_anomaly),
+                           mean_motion, np.radians(raan))
+    else:
+        # Otherwise, use TLEs
+        satellite = sapi.Satrec.twoline2rv(line1, line2, sapi.WGS72)
+        mean_motion = satellite.mm
+
+    if one_orbit:
+        ind = times <= (2 * np.pi / mean_motion * 60)
+        times = times[ind]
+        index = index[ind]
+
+    jd = jd * np.ones(len(times))
+    fr = times / 86400.
+
+    err_code, position, velocity = satellite.sgp4_array(jd, fr)
+
+    # Check all propagated values for errors in propagation
+    errors = np.unique(err_code[err_code > 0])
+    if len(errors) > 0:
+        # Raise highest priority error.
+        raise ValueError(sapi.SGP4_ERRORS[errors[0]])
+
+    # Add ECEF values to instrument.
+
+    pos_ecef = conv_sph.eci2ecef(position, (jd + fr))
+    vel_ecef = conv_sph.eci2ecef(velocity, (jd + fr))
+
+    # Convert to geocentric latitude, longitude, altitude.
+    lat, lon, rad = conv_sph.ecef_cart2spherical(pos_ecef)
+    # Convert to geodetic latitude, longitude, altitude.
```

Comment:
```suggestion

        # Convert to geodetic latitude, longitude, altitude.
```
I was previously directed to have new line before comment/code blocks